lsp/source/lib/util.mts

```diff
@@ -401,3 +401,21 @@ function diagnosticCategoryToSeverity(
       return DiagnosticSeverity.Information;
   }
 }
+
+/**
+ * https://github.com/tc39/proposal-promise-with-resolvers
+ */
+export function withResolvers<T>() {
+  let resolve: (value: T) => void;
+  let reject: (reason?: any) => void;
+  const promise = new Promise<T>((res, rej) => {
+    resolve = res;
+    reject = rej;
+  });
+  return { promise, resolve: resolve!, reject: reject! };
```

Comment:
```suggestion
  return { promise, resolve!, reject! };
```

Reply:
Oops, never mind, this is TypeScript, not Civet. 😢

Reply:
> Oops, never mind, this is TypeScript, not Civet. 😢

In fact, at the very beginning, I also wrote it this way. 😢